File: 프로그래머스/unrated/132267. 콜라 문제/콜라 문제.java
class Solution {
    public int solution(int a, int b, int n) {
        
    /*
    3개의 경우
    20 / 3 = 6, 20 % 3 = 2
    6 + 2 = 8(가지고 있는 콜라병 수)
    
    8 / 3 = 2, 8 % 3 = 2
    2 + 2 = 4
    
    4 / 3 = 1, 4 % 3 = 1
    1 + 1 = 2
    
    3 < 2 !
    
    a = 4 
    
    
    */
        int answer = 0;
        
        while (n >= a) {
            int takenCola = (n / a) * b;
            int myCola = n % a;
            answer += takenCola;
            n = takenCola + myCola;
        }
        return answer;
    }
}

File: 프로그래머스/1/132267. 콜라 문제/콜라 문제.py
def solution(a, b, n):
    answer, gainNum, remainNum = 0, 0, 0
    
    
    
    while n > a:
        answer += (n // a) * b
        remainNum = n % a
        n = (n // a) * b + remainNum
        
    answer += (n // a) * b
    
    return answer

# 20개

# 6개받고  2개남음
